src/NuGet.Clients/PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -81,7 +195,7 @@ public UIActionEngine(ISourceRepositoryProvider sourceProvider, NuGetPackageMana
                 if (!token.IsCancellationRequested)
                 {
                     // execute the actions
-                    await ExecuteActionsAsync(actions, uiService.ProgressWindow, userAction, token);
+                    await executeActionsTask(actions);
```

Comment:
I think we should add the Async suffix for Func parameters that returns a Task. So that await always happens on things ending with async. Please do it here and everywhere else in the PR